Framework/Core/include/Framework/AnalysisDataModel.h

```diff
@@ -61,6 +61,30 @@ DECLARE_SOA_DYNAMIC_COLUMN(Phi, phi, [](float snp, float alpha) -> float { retur
 DECLARE_SOA_DYNAMIC_COLUMN(Eta, eta, [](float tgl) -> float { return log(tan(0.25 * M_PI - 0.5 * atan(tgl))); });
 DECLARE_SOA_DYNAMIC_COLUMN(Pt, pt, [](float signed1Pt) -> float { return fabs(1.0 / signed1Pt); });
 DECLARE_SOA_DYNAMIC_COLUMN(Charge, charge, [](float signed1Pt) -> short { return (signed1Pt > 0) ? 1 : -1; });
+DECLARE_SOA_DYNAMIC_COLUMN(Px, px, [](float signed1Pt, float snp, float alpha) -> float {
+  double pt = 1. / std::abs(signed1Pt);
+  double cs = std::cos(alpha);
+  double sn = std::sin(alpha);
+  double r = std::sqrt((1. - snp) * (1. + snp));
+  return pt * (r * cs - snp * sn);
+});
```

Comment:
Can these columns already depend on each other? In this case I would suggest to use directly pt as input to this one, not signed1Pt

Reply:
for the moment it's not possible.